### PR TITLE
Updated channel and documentation to Rubocop v0.80.0

### PR DESCRIPTION
**Pull request by DanielWright**
Most of this PR is pretty straightforward, and I'll leave some PR comments where I've got questions. At a minimum, this branch should provide a solid base for updating the channel.

### CODE REVIEW

**Comment by davehenton** [comment]
@filipesperandio

File: spec/cc/engine/config_upgrader_spec.rb [hunk 1]
```diff
@@ -17,6 +17,12 @@ def get_true
           Enabled: false
         Style/FrozenStringLiteralComment:
           Enabled: false
+        Style/HashEachMethods:
+          Enabled: false
+        Style/HashTransformKeys:
+          Enabled: false
+        Style/HashTransformValues:
+          Enabled: false
```
Comment (DanielWright) [inline spec/cc/engine/config_upgrader_spec.rb]: While this change does cause the spec to pass, I'm not sure it's right way to handle the stdout warning about new cops.

**Comment by DanielWright** [comment]
I should also note that the RSpec suite does run, but with some additional noise in the output from the following Rubocop warning:

```
The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file:
 - Style/HashEachMethods
 - Style/HashTransformKeys
 - Style/HashTransformValues
```

**Comment by andyw8** [comment]
@DanielWright that's expected due to https://github.com/rubocop-hq/rubocop/pull/7567

**Comment by DanielWright** [comment]
Is it desirable to handle or suppress the warnings in this PR?

**Comment by andyw8** [comment]
I guess the Code Climate team need to make a decision here on whether upgrading the channel should make the newer cops available by default or not.

File: spec/cc/engine/config_upgrader_spec.rb (hunk 1 above)
Comment (filipesperandio) [reply spec/cc/engine/config_upgrader_spec.rb]: I would say let's turn them ON!

Comment (DanielWright) [reply spec/cc/engine/config_upgrader_spec.rb]: Your wish is my command! 51b5aa4

**Comment by andyw8** [comment]
cc @pirj @bbatsov in case of any concerns about this.

**Comment by bbatsov** [comment]
No concerns here. As state earlier, we switched to a policy where RuboCop will enable cops only at major releases and the rest of the time it'd be up to users to decide if they want to enable/disable some new cop. I view the message you mentioned as purely informational, not a warning. :-)

File: spec/cc/engine/config_upgrader_spec.rb (hunk 1 above)
Comment (pean) [reply spec/cc/engine/config_upgrader_spec.rb]: Makes sense to enable them since they are enabled by default in Rubocop.

Comment (bbatsov) [reply spec/cc/engine/config_upgrader_spec.rb]: Note that they are not enabled by default in RuboCop, but rather "pending" which means the users have to decide whether to enable them or not. We'll likely enable them by default in version 1.0.

File: config/contents/style/block_delimiters.md [hunk 2]
```diff
@@ -100,3 +100,29 @@ multi-line blocks.
...
+### Example: BracesRequiredMethods: ['sig']
+
+    # Methods listed in the BracesRequiredMethods list, such as 'sig'
+    # in this example, will require `{...}` braces. This option takes
+    # precedence over all other configurations except IgnoredMethods.
+
+    # bad
+    sig do
+      params(
+        foo: string,
```
Comment (pirj) [inline config/contents/style/block_delimiters.md]: Trailing comma, is this a deliberate choice?

File: spec/cc/engine/config_upgrader_spec.rb (hunk 1 above)
Comment (filipesperandio) [reply spec/cc/engine/config_upgrader_spec.rb]: Ok, lets leave it on in the spec, but keeping it out of the default config, so it warns users to make a decision, just like rubocop itself.

Comment (filipesperandio) [reply spec/cc/engine/config_upgrader_spec.rb]: Ok, lets leave it on in the spec, but keeping it out of the default config, so it warns users to make a decision, just like rubocop itself.

File: config/contents/style/block_delimiters.md (hunk 2 above)
Comment (DanielWright) [reply config/contents/style/block_delimiters.md]: This appears to be [a choice made upstream](https://github.com/rubocop-hq/rubocop/blob/v0.80.0/lib/rubocop/cop/style/block_delimiters.rb#L119), whether a typo or deliberate I cannot say.

File: spec/cc/engine/config_upgrader_spec.rb (hunk 1 above)
Comment (DanielWright) [reply spec/cc/engine/config_upgrader_spec.rb]: Done in a81a64e. ✅

Comment (filipesperandio) [reply spec/cc/engine/config_upgrader_spec.rb]: Github is not helping much today... seems to be having issues.
I've tried to merge this early today and seems like the branch got into a weird state instead.

... just checked, it actually merged the changes in, but kept the PR open! So weird.

Comment (filipesperandio) [reply spec/cc/engine/config_upgrader_spec.rb]: The diff is now weird, but your changes are there.
I will be releasing this soon.

**Pull request merged**